File: Core/Models/Games/MancalaGameState.cs
using Core.Models.Interfaces;

namespace Core.Models.Games
{
    public class MancalaGameState : IGameState
    {
        public bool HasGameBeenSetup { get; set; }
        public bool GameIsPlayable { get; set; }
        public bool IsPlayer1Turn { get; set; }
        public MancalaPlayer Player1 { get; set; }
        public MancalaPlayer Player2 { get; set; }
        public string Result { get; set; }
    }
}

File: UI.Web/ClientApp/src/app/models/MancalaGameState.ts
import { MancalaPlayer } from "./MancalaPlayer";

export class MancalaGameState{
    constructor(){    }
    
    hasGameBeenSetup: boolean;
    gameIsPlayable: boolean;
    isPlayer1Turn: boolean;
    player1: MancalaPlayer;
    player2: MancalaPlayer;
}